workflow/controller/operator.go

```diff
@@ -589,13 +589,18 @@ func (woc *wfOperationCtx) persistUpdates(ctx context.Context) {
 	// errors if we label a pod that was deleted already.
 	for podName, podPhase := range woc.completedPods {
 		if woc.execWf.Spec.PodGC != nil {
+			var doPodGC bool
```

Comment:
`do*` variables are kind of smelly, they tell your cause and effect have been separated in the code and risk bugs, instead just inline `queuePodForCleanupAfter` please

Reply:
Okies will do, I had copied the pattern from elsewhere in the code but happy to oblige!